test/ecto/model/dependent_test.exs

```diff
@@ -0,0 +1,121 @@
+Code.require_file "../../support/mock_repo.exs", __DIR__
```

Comment:
Let's drop those unit tests, they are not doing anything really, in favor of integration tests. This way we can drop the changes to mock_repo too. :)